### DISCUSSION TITLE
v4.0.0-beta

### DISCUSSION THREAD

**Discussion by danielmorell** [rollbar/rollbar-php#597] [General]
This beta release of 4.0.0 fixes the psr/log v3.0 issue as well as adds a lot of typing improvements to the public interfaces and methods.

With the update to psr/log v3 `Rollbar::log()`, and other logging methods defined in the psr/log interface now return `void`. If you still need the `Response` return value from those methods, you can now use the new `Rollbar::report()` method which returns `Response`.

### Added
* PHP 8 language level mitigations, add typehints by @Chris8934 in #569.
* Added support for `psr/log` v3 by @danielmorell in #577.
* Added the `Rollbar::report()` and `RollbarLogger::report()` methods to continue to support `Response` return type requirements after upgrading to psr/log v3 by @danielmorell in #577.
* Added comments and type annotations to the `EncodedPayload` class and payload interfaces by @danielmorell in #581.
* Added typing / comments to `Rollbar` and `RollbarLogger` classes by @danielmorell in #585.
* Added required public methods to the `DataBuilderInterface` by @danielmorell in #586.
* Added typing / comments to the `ResponseHandlerInterface` by @danielmorell in #588.
* Added typing / comments to the `ScrubberInterface` and `Scrubber` class by @danielmorell in #591.
* Added typing / comments to the `FilterInterface` by @danielmorell in #587.
* Added typing / comments to the `SenderInterface` by @danielmorell in #592.

### Changed
* Renamed `IStrategy` to `StrategyInterface` updated `Truncation` and changed custom truncation strategy from requiring class extend the `AbstractStrategy` to now require it implement `StrategyInterface` by @danielmorell in #580.
* Replaced the `FilterInterface::shouldSend()` `$accessΤoken` argument with `$isUncaught` making it close to `check_ignore` usage @danielmorell in #587.

### Removed
* Removed deprecated log levels and fixed inconsistent use of `Rollbar/LevelFactory` by @danielmorell in #578.
* Removed previously deprecated reporting methods from `Rollbar` by @danielmorell in #579.
* Removed the `null` return type from `TransformerInterface::getPayload()` by @danielmorell in #593.

### Fixed
* Fixed call of method name changed in 8fac418 by @danielmorell in #583. 
* Fixed #461 Added support for `psr/log` context exception by @danielmorell in #582.
* Fixed #469 Added `requireAccessToken()` method to `SenderInterface` by @danielmorell in #595.

## New Contributors
* @paulserraino made their first contribution in https://github.com/rollbar/rollbar-php/pull/584

**Full Changelog**: https://github.com/rollbar/rollbar-php/compare/v3.1.4...v4.0.0-beta

<hr /><em>This discussion was created from the release <a href='https://github.com/rollbar/rollbar-php/releases/tag/v4.0.0-beta'>v4.0.0-beta</a>.</em>

**Comment by adantart** [comment]
First of all... Congrats!!

I've installed the 4.0 and have an issue (maybe my fault, too late here :-D )

If I use in this way it works:
```
\Rollbar\Rollbar::init([
    'access_token' => 'xxxxxxxxxxxxxxxxxxxxxxx', 
    'environment'  => 'development',
]);
$log = new \MonoLog\Logger('general');
$log->pushHandler(new \Monolog\Handler\RollbarHandler(\Rollbar\Rollbar::logger()), \MonoLog\Logger::DEBUG);
$log->error('Test');
```

But if I use through log, it does not work:

I added in config/logging.php:
```
...
'default' => 'stack',
...
'channels' => [
        'stack' => [
            'driver' => 'stack',
            'channels' => ['single', 'rollbar'],
            'ignore_exceptions' => false,
        ],
...

        'rollbar' => [
            'driver' => 'monolog',
            'handler' => \Rollbar\Laravel\MonologHandler::class,
            'access_token' => env('ROLLBAR_TOKEN'),
            'level' => 'debug',
        ],

...
```

and added the ROLLBAR_TOKEN in the .env, of course ;-)

But when I use this:
\Log::debug('Test debug message');

it does not trigger the rollbar channel.

Any ideas ?

THANK YOU in advance

**Reply by adantart** [reply]
Ok, my mistake ... my misunderstanding ... but I write here in case somebody needs it ;-)

Don't install this 4.0.0-beta if you want to use with Laravel... what you have to do is to download the rollbar laravel package (https://github.com/rollbar/rollbar-php-laravel) ... modify composer and put:
```
"rollbar/rollbar": "^2.0 | ^3.1 | ^4.0"
```

it will take this package from repo correctly ;-)

**Comment by adantart** [comment]
I've been testing in a "pre-live" environment for 2 weeks, and works fine!
Ready to publish it officially ? ;-)

**Comment by adantart** [comment]
Any idea of publishing date ?
I think it has passed a long testing time ;-)

**Reply by danielmorell** [reply]
I am hoping to have the stable out on Monday the 20th.
